libs/jwt-tools/src/Data/Jwt/Tools.hs

```diff
@@ -198,7 +198,10 @@ generateDpopToken dpopProof uid cid domain nonce uri method maxSkewSecs maxExpir
     toResult _ _ = Left FfiError
 
     toCStr :: forall a m. (ToByteString a, MonadIO m) => a -> m CString
-    toCStr = liftIO . newCString . cs . toByteString'
+    toCStr = liftIO . newCString . toStr
+      where
+        toStr :: a -> String
+        toStr = cs . toByteString'
```

Comment:
`newCString` takes a `String`?!  that's horrible.  :')

but ok, then this is fine i suppose...  you could have also left things inline and written `(cs @ByteString @String)` instead of `cs`.